kvbc/src/categorization/kv_blockchain.cpp

```diff
@@ -825,6 +825,16 @@ std::optional<Hash> KeyValueBlockchain::parentDigest(BlockId block_id) const {
   return block_chain_.parentDigest(block_id);
 }
 
+std::optional<Hash> KeyValueBlockchain::calculateBlockDigest(BlockId block_id) const {
+  // get the block from ST chain or block chain.
+  auto raw_block = getRawBlock(block_id);
+  if (raw_block) {
```

Comment:
raw_block is not a pointer?

Reply:
Thanks @eyalrund , yes you are correct raw_block is not a pointer, its a optional.